Dockerfile

```diff
@@ -51,17 +53,35 @@ COPY .gitconfig /.gitconfig
 
 WORKDIR /CIRRUS-core
 
+# Python310 target
+FROM core_base AS python310
+RUN \
+        dnf groupinstall "Development Tools" -y && \
+        dnf install openssl-devel bzip2-devel libffi-devel -y && \
+        cd /usr/src && \
+        wget https://www.python.org/ftp/python/${PYTHON_3_10_VERSION}/Python-${PYTHON_3_10_VERSION}.tgz && \
+        tar xzf Python-${PYTHON_3_10_VERSION}.tgz && cd Python-${PYTHON_3_10_VERSION} && ./configure --enable-optimizations && \
```

Comment:
BTW I have found in the past that when installing from source, the default configuration is quite a bit different from what you would get from a package distribution such as apt. Mainly, it creates a statically linked executable which can break certain C-extensions that rely on being able to dynamically link to python. There's also some sqlite3 features that are not enabled by default that can cause issues if you're trying to use (or a package you depend on tries to use) the builtin `sqlite3` package.

I found in the past that installing it like this solves those issues:
```
./configure --enable-optimizations \
              --enable-shared \
              --enable-loadable-sqlite-extensions \
              --prefix /usr/local \
              LDFLAGS=-Wl,-rpath=/usr/local/lib
```

The `--prefix` and `LDFLAGS` supposedly can be optional but I just include them because I have found that guarantees that everything works right. If you want to install somewhere other than `/usr/local` then obviously you have to change those two arguments.

Reply:
Also here is how the official Python docker container builds python:

https://github.com/docker-library/python/blob/9602fc7493ee0d09c124a05553aadee142992df8/3.8/slim-bullseye/Dockerfile#L71-L78

Reply:
For sqlite you also have to install the sqlite-dev package to get the c headers.